test/hotkey/counter_table/check_counter_table.c

```diff
@@ -0,0 +1,99 @@
+#include <hotkey/counter_table.h>
+
+#include <hotkey/constant.h>
+
+#include <check.h>
+
+#include <stdlib.h>
+
+#define SUITE_NAME "counter_table"
+#define DEBUG_LOG  SUITE_NAME ".log"
+
+#define TEST_TABLE_SIZE 10
+
+/*
+ * utilities
+ */
+static void
+test_setup(void)
+{
+    counter_table_setup(TEST_TABLE_SIZE, TEST_TABLE_SIZE);
+}
+
+static void
+test_teardown(void)
+{
+    counter_table_teardown();
+}
+
+static void
+test_reset(void)
+{
+    test_teardown();
+    test_setup();
+}
+
+/**************
+ * test cases *
+ **************/
+
+START_TEST(test_basic)
+{
+    char *key1 = "key1", *key2 = "key2";
+    uint32_t klen = 4;
```

Comment:
Use two keys with different length. Also, since the char* is actually 5 bytes as it has a NULL termination, use 5 instead of 4 in this case, as it may help catch off-by-one errors.